src/Parsers/ASTSetQuery.cpp

```diff
@@ -81,10 +81,30 @@ void ASTSetQuery::formatImpl(WriteBuffer & ostr, const FormatSettings & format,
             first = false;
 
         formatSettingName(change.name, ostr);
-        CustomType custom;
-        if (!format.show_secrets && change.value.tryGet<CustomType>(custom) && custom.isSecret())
-            ostr << " = " << custom.toString(false);
-        else
+
+        auto format_if_secret = [&]() -> bool
+        {
+            CustomType custom;
+            if (change.value.tryGet<CustomType>(custom) && custom.isSecret())
+            {
+                ostr << " = " << custom.toString(/* show_secrets */false);
+                return true;
+            }
+
+            if (state.create_engine_name == "Iceberg")
```

Comment:
My only issue with this is it is really easy to change the engine name while missing to change this one. As this is a security motivated change, I think it would be nice to make it safer. However I know that we cannot linkd `dbms` to parsers, and specifying a constant in the parser just because of this seems silly. It also doesn't solve the issue of introducing/renaming settings or just adding new ones.

All in all I don't know how to make it stricter, but maybe it worth to add a comment to DatabaseIceberg and the settings about this quirk, so at least if somebody reads the comment while adding/renaming a setting or renaming the database, they can catch this.

Reply:
Also for engine arguments it is done in a similar way https://github.com/ClickHouse/ClickHouse/blob/d99795b7636b6542a860ef2144283943d9f882f8/src/Parsers/FunctionSecretArgumentsFinder.h#L103-L108
so I guess it should be ok as is, as we already do this